### PR TITLE
[ci] Add LUCI version of macOS custom tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
This adds a bringup LUCI version of the macOS-host custom package tests. This is the first attempt to migrate a flutter/packages or flutter/plugins test to macOS ARM, so is experimental.

The motivation for doing this migration now is to see if it can unblock https://github.com/flutter/packages/pull/2816, which doesn't seem to be able to run successfully on Cirrus ARM machines (very possibly due to the same issue that blocked flutter/plugins migrations to ARM on Cirrus).

### CODE REVIEW

**Comment by stuartmorgan** [comment]
(If this doesn't end up working, I'll try a LUCI Intel version of this test instead to see if that unblocks the iOS pigeon PR.)

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:69]: Did you try a `led` run with this?  Suspect you'll hit https://github.com/flutter/flutter/issues/113231

Comment (jmagman) [reply .ci.yaml:69]: Technically you can copy an existing config and edit it go/flutter-luci-recipes#testing-a-local-change-for-a-new-builderrecipe.  However I've never had the patience to do that.
Usually I remove bringup and presubmit in the PR and let it run in presubmit, and then revert that before merging.  ci.yaml validation would fail though.  You can normally be sneaky and use an existing test and just say swap out `target_file` to have it run in presubmit, but you have the added complication of there being no existing `Mac_arm64` builders.

**Review by jmagman — APPROVED** [review]
Change LGTM though I suspect this may fail due to arm infra issues.  Worth a shot though.

File: .ci.yaml
Comment (stuartmorgan) [reply .ci.yaml:69]: Yeah, that sounds about like what I've done in the past; this was to bootstrap me to something where I could play around with things in presubmit in a follow-up PR.

But since we have known ARM breakage I'll hold off on this, and maybe explore doing an Intel version instead. Thanks for the pointer to that issue!

**Comment by stuartmorgan** [comment]
Re-opening to try again now that https://github.com/flutter/flutter/issues/113231 is fixed.

**Pull request merged**
